### ISSUE TITLE
Dylint crashes on MacOS Sonoma

### ISSUE THREAD

**Issue by tjade273** [trailofbits/dylint#857]
```
> rustup show
Default host: aarch64-apple-darwin
rustup home:  /Users/tjaden/.rustup

installed toolchains
--------------------

stable-aarch64-apple-darwin
nightly-2022-01-05-aarch64-apple-darwin
nightly-2022-10-28-aarch64-apple-darwin
nightly-2023-03-10-aarch64-apple-darwin
nightly-2023-04-23-aarch64-apple-darwin
nightly-2023-08-12-aarch64-apple-darwin
nightly-2023-08-24-aarch64-apple-darwin
nightly-aarch64-apple-darwin
1.54-aarch64-apple-darwin
1.65-aarch64-apple-darwin (default)
1.56.1-aarch64-apple-darwin
1.66.0-aarch64-apple-darwin

active toolchain
----------------

1.65-aarch64-apple-darwin (default)
rustc 1.65.0 (897e37553 2022-11-02)

> cargo dylint

*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString stringByStandardizingPath]: unrecognized selector sent to instance 0x6000012c0400'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000189b608c0 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x0000000189659eb4 objc_exception_throw + 60
	2   CoreFoundation                      0x0000000189c123dc -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x0000000189aca9e4 ___forwarding___ + 1572
	4   CoreFoundation                      0x0000000189aca300 _CF_forwarding_prep_0 + 96
	5   Foundation                          0x000000018ab9f4f0 -[NSProcessInfo arguments] + 188
	6   CoreFoundation                      0x0000000189bdbff8 __getDefaultArguments_block_invoke + 96
	7   libdispatch.dylib                   0x0000000189861910 _dispatch_client_callout + 20
	8   libdispatch.dylib                   0x000000018986314c _dispatch_once_callout + 32
	9   CoreFoundation                      0x0000000189bdb994 _addBackstopValuesForIdentifierAndSource + 640
	10  CoreFoundation                      0x0000000189a96254 __81-[_CFXPreferences(SourceAdditions) withNamedVolatileSourceForIdentifier:perform:]_block_invoke + 144
	11  CoreFoundation                      0x0000000189bdb63c -[_CFXPreferences withNamedVolatileSourceForIdentifier:perform:] + 272
	12  CoreFoundation                      0x0000000189a9c5c4 -[CFPrefsSearchListSource addNamedVolatileSourceForIdentifier:] + 136
	13  CoreFoundation                      0x0000000189c1a8bc __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke.155 + 296
	14  CoreFoundation                      0x0000000189c1a564 -[_CFXPreferences withSearchLists:] + 84
	15  CoreFoundation                      0x0000000189a97b18 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 300
	16  CoreFoundation                      0x0000000189c1a710 -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 384
	17  CoreFoundation                      0x0000000189a97440 -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 156
	18  CoreFoundation                      0x0000000189a97368 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 112
	19  SystemConfiguration                 0x000000018a827478 SCDynamicStoreCopyProxiesWithOptions + 180
	20  cargo-dylint                        0x0000000100db9948 Curl_macos_init + 16
	21  cargo-dylint                        0x0000000100da7d10 curl_global_init + 176
	22  cargo-dylint                        0x0000000100d9538c _ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h12453bd12903a199E + 36
	23  cargo-dylint                        0x000000010110b75c _ZN3std10sys_common4once5queue4Once4call17hbda0067e1c327d63E + 1132
	24  cargo-dylint                        0x0000000100d958e0 _ZN4curl9INIT_CTOR9init_ctor17hc7ee4fb9417f2714E + 88
	25  dyld                                0x00000001896b15c8 ___ZZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateEENK3$_0clEv_block_invoke + 168
	26  dyld                                0x00000001896f6920 ___ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv_block_invoke.209 + 340
	27  dyld                                0x00000001896e9c60 ___ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE_block_invoke + 496
	28  dyld                                0x00000001896912fc _ZNK5dyld39MachOFile18forEachLoadCommandER11DiagnosticsU13block_pointerFvPK12load_commandRbE + 300
	29  dyld                                0x00000001896e8c98 _ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE + 192
	30  dyld                                0x00000001896f6434 _ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv + 516
	31  dyld                                0x00000001896ad798 _ZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateE + 448
	32  dyld                                0x00000001896b3b14 _ZNK5dyld416JustInTimeLoader15runInitializersERNS_12RuntimeStateE + 36
	33  dyld                                0x00000001896adb4c _ZNK5dyld46Loader23runInitializersBottomUpERNS_12RuntimeStateERN5dyld35ArrayIPKS0_EE + 220
	34  dyld                                0x00000001896b1654 _ZZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateEENK3$_1clEv + 112
	35  dyld                                0x00000001896adccc _ZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateE + 304
	36  dyld                                0x00000001896d2ad4 _ZN5dyld44APIs25runAllInitializersForMainEv + 464
	37  dyld                                0x0000000189695f34 _ZN5dyld4L7prepareERNS_4APIsEPKN5dyld313MachOAnalyzerE + 3192
	38  dyld                                0x0000000189694f44 start + 1948
)
libc++abi: terminating due to uncaught exception of type NSException
[1]    23773 abort      cargo dylint
```

**Comment by smoelius** [comment]
Looks suspiciously like this: https://github.com/curl/curl/issues/11893

**Comment by smoelius** [comment]
Unfortunately, I don't have good workaround right now. But I will try to confirm that as the issue, and pursue a fix upstream.

**Comment by smoelius** [comment]
@tjade273 Could I trouble you to try this, and then repeat your previous experiment?
```
cargo install cargo-dylint --git https://github.com/trailofbits/dylint --branch curl-sys
```
`curl-sys` is the "merge from" branch of this PR: https://github.com/trailofbits/dylint/pull/858

**Comment by tjade273** [comment]
That seems to have fixed it

**Comment by smoelius** [comment]
Fixed by #858. Thanks for reporting and for testing out the fix, @tjade273. 🙏
